chore: create performance metrics project

DerekMaggio opened the pull request:
# Overview

Basic setup of Performance Metrics project

Closes https://opentrons.atlassian.net/browse/EXEC-380

# Test Plan

- [x] Verify all makefile commands can run successfully (GH Actions will be in another PR)
- [x] Build wheel, install, and verify it runs

DerekMaggio committed:
chore: create project

sfoster1 reviewed (changes requested):
Let's keep the tooling we use in monorepo python projects consistent, with flake8, mypy, black, and so on at the same versions as the other projects. This eases everybody's ability to context switch.

DerekMaggio committed:
chore: standardize to other projects in monorepo

sfoster1 commented on performance-metrics/pyproject.toml:
```diff
@@ -0,0 +1,13 @@
+[tool.ruff]
```
need to remove this one too - we're still using `setup.py`

DerekMaggio replied:
https://github.com/Opentrons/opentrons/pull/14806/commits/71cb58415df045e67a42063338f3080048836b43 should fix it

DerekMaggio committed:
chore: switch over to setup.py

formatting

chore: formatting and configs

chore: remove comment add more cleaning

fix data

DerekMaggio committed:
Use shortcuts

sfoster1 reviewed (approved):
lgtm!
